src/main/java/uk/gov/hmcts/darts/annotation/service/impl/AnnotationDataManagement.java

```diff
@@ -53,20 +54,11 @@ public Map<ExternalLocationTypeEnum, UUID> upload(BinaryData binaryData, String
     }
 
     public InputStreamResource download(List<ExternalObjectDirectoryEntity> externalObjectDirectoryEntities) {
-        var downloadableExternalObjectDirectories = DownloadableExternalObjectDirectories
-            .getFileBasedDownload(externalObjectDirectoryEntities);
-
-        dataManagementFacade.getDataFromUnstructuredArmAndDetsBlobs(downloadableExternalObjectDirectories);
-        var downloadResponseMetaData = downloadableExternalObjectDirectories.getResponse();
-
         try {
-            if (!downloadResponseMetaData.isSuccessfulDownload()) {
-                downloadResponseMetaData.close();
-                throw new DartsApiException(FAILED_TO_DOWNLOAD_ANNOTATION_DOCUMENT);
-            }
+            DownloadResponseMetaData downloadResponseMetaData = dataManagementFacade.retrieveFileFromStorage(externalObjectDirectoryEntities);
 
             return new InputStreamResource(downloadResponseMetaData.getInputStream());
-        } catch (IOException e) {
+        } catch (IOException | FileNotDownloadedException e) {
             log.error("Failed to download annotation document {}",
                       externalObjectDirectoryEntities.get(0).getAnnotationDocumentEntity().getId(), e);
```

Comment:
I think we still need to do a `downloadResponseMetaData.close();` here as we're not using the try-with resources on `DownloadResponseMetaData`?
We want the InputStream for the file attachment.
@mestebanez knows more, but there was talk about needing to cleanup the file later when Spring closes the InputStream.

Reply:
Would that not be out of scope of this ticket?

Reply:
using try-with-resources now instead.

Reply:
I'm not sure we can though as I think it would break the annotation / transcription download endpoints?
Please can you check that still works with your latest code

Reply:
reverted